lib/active_model/serializer/attributes.rb

```diff
@@ -0,0 +1,109 @@
+module ActiveModel
+  class Serializer
+    module Attributes
+      class Attribute
+        delegate :call, to: :reader
+
+        attr_reader :name, :reader
+
+        def initialize(name)
+          @name = name
+          @reader = :no_reader
+        end
+
+        def self.build(name, block)
+          if block
+            AttributeBlock.new(name, block)
+          else
+            AttributeReader.new(name)
+          end
+        end
+      end
+      class AttributeReader < Attribute
+        def initialize(name)
+          super(name)
+          @reader = ->(instance) { instance.read_attribute_for_serialization(name) }
+        end
+      end
+      class AttributeBlock < Attribute
+        def initialize(name, block)
+          super(name)
+          @reader = ->(instance) { instance.instance_eval(&block) }
+        end
+      end
```

Comment:
Would you mind showing me what you have in mind? I think this is 90% a style thing, at this point, and I just found this more readable and extendable.  Having the `Attribute` object let me keep track of the name, which I needed to use, which is why I introduced it (as opposed to just wanting it there for a future need)